File: spring-data-requery/src/main/java/org/springframework/data/requery/provider/ProxyIdAccessor.java
package org.springframework.data.requery.provider;

import org.jetbrains.annotations.Nullable;

/**
 * org.springframework.data.requery.provider.ProxyIdAccessor
 *
 * @author debop
 * @since 18. 6. 7
 */
public interface ProxyIdAccessor {

    boolean shouldUseAccessorFor(Object entity);

    @Nullable
    Object getIdentifierFrom(Object entity);
}

File: spring-data-requery/src/main/kotlin/org/springframework/data/requery/provider/ProxyIdAccessor.kt
package org.springframework.data.requery.provider

/**
 * org.springframework.data.requery.provider.ProxyIdAccessor
 *
 * @author debop
 */
interface ProxyIdAccessor {

    fun shouldUseAccessorFor(entity: Any?): Boolean

    fun getIdentifierFrom(entity: Any?): Any?
}